### ISSUE TITLE
`manual_let_else` doesn't trigger if binding has a type annotation.

### ISSUE THREAD

**Issue by ghost** [rust-lang/rust-clippy#10150]
### Summary

`manual_let_else` doesn't trigger if binding has a type annotation.

### Lint Name

manual_let_else 

### Reproducer

I tried this code:

```rust
#![warn(clippy::pedantic)]
#![allow(unused_variables)]

fn main() {
    let x: u32 = match "14".parse() {
        Ok(i) => i,
        Err(_) => return,
    };
}
```

I expected to see this happen:
A suggestion to use:
```rust
let Ok(i): Result<u32, _> = "14".parse() else { return };
```

Instead, this happened:
Nothing. This is a false negative.

### Version

```text
clippy: 0.1.68 (2023-01-02 d6f99e5)
```

**Comment by EricWu2003** [comment]
hmmm, it looks like this is behavior intended when the `manual_let_else` lint was created (see [this commit](https://github.com/rust-lang/rust-clippy/commit/748169deaa3c57e15e3a84bb90e033dddaa64e03)). We don't perform the lint because expressing the type might be too difficult.

@est31 might be able to provide some more context here about why the lint doesn't trigger when the type annotation is present.

**Comment by est31** [comment]
The reason why I've made 748169deaa3c57e15e3a84bb90e033dddaa64e03 is because there is no type ascription, so you have to name the full type. Say it's code like:

```Rust
let b: String = if let Foo { bar: Bar { b, .. }, .. } = get_foo() { b } else { panic!() };
```

Then, to specify the type as `String` in a let-else construct, you have to do:

```Rust
let Foo { bar: Bar { b, .. }, .. }: Foo<Bar<String>> = get_foo() else { panic!() };
```

or some other arbitrarily deep change of types. The resulting code might not look nicer than before, so I went with not firing the lint. Of course suggestions make it even harder.
